src/test_workflow/benchmark_test/benchmark_test_cluster.py

```diff
@@ -132,8 +134,12 @@ def setup_cdk_params(self, config: dict) -> dict:
             artifact_url = self.manifest.build.location if isinstance(self.manifest, BundleManifest) else \
                 f"https://artifacts.opensearch.org/snapshots/core/opensearch/{self.manifest.build.version}/opensearch-min-" \
                 f"{self.manifest.build.version}-linux-{self.manifest.build.architecture}-latest.tar.gz"
+            if not self.args.insecure and semver.compare(self.manifest.build.version, '2.12.0') != -1:
+                need_strong_password = True
         else:
             artifact_url = self.args.distribution_url.strip()
+            if not self.args.insecure and semver.compare(self.args.distribution_version, '2.12.0') != -1:
+                need_strong_password = True
```

Comment:
Since both if and else conditions needs these lines of code maybe move it to line 136?

Reply:
Did not get this. This condition is evaluated based on whether manifest has been provided or distribution url + version is provided.

Reply:
Optimization here wouldn't really yield better results to I kept it simple.